lib/expression/callbacks.ex

```diff
@@ -93,12 +93,25 @@ defmodule Expression.Callbacks do
       function_exported?(Standard, vargs_function_name, 2) ->
         {:vargs, Standard, vargs_function_name, 2}
 
+      # Check if the wrong number of arguments was provided
+      wrong_arity_but_function_exists?(module, exact_function_name) ->
+        {:error, "wrong number of arguments to #{function_name}."}
+
+      # Check if the wrong number of arguments was provided
+      wrong_arity_but_function_exists?(Standard, exact_function_name) ->
+        {:error, "wrong number of arguments to #{function_name}."}
+
       # Otherwise fail
       true ->
         {:error, "#{function_name} is not implemented."}
     end
   end
 
+  defp wrong_arity_but_function_exists?(module, function_name)
+       when is_atom(module) and is_atom(function_name) do
+    Enum.any?(0..20, fn arity -> Kernel.function_exported?(module, function_name, arity) end)
```

Comment:
I like the pragmatism here and I've not actually done this myself before but perhaps [`Module.__info__()[:functions]`](https://hexdocs.pm/elixir/Module.html#c:__info__/1) is a way to do this? The docs suggest that returns a list of functions and their arities so maybe that's worth cross referencing?

Reply:
That's much better, thank you!

TBH the `Enum.any?(0..20, ...` was a ChatGPT suggestion - I'm still an Elixir newbie so great to learn this!